types/src/receipt/transaction_receipt.rs

```diff
@@ -1,16 +1,16 @@
-use alloy_rlp::RlpEncodable;
+use alloy_rlp::{BufMut, BytesMut, Encodable};
+use serde::{Deserialize, Serialize};
 
-use crate::Bloom;
+use crate::{Bloom, Log};
 
-use super::log::Log;
 use super::tx_type::TxType;
 
 /// The receipt structure containing logs from smart contracts we are listening to; adapted from
 /// [`reth_primitives::ReceiptWithBloom`][1].
 ///
 /// [1]: https://ethereum.org/en/developers/docs/blocks/#block-anatomy
 /// [2]: https://github.com/paradigmxyz/reth/blob/f41386d28e89dd436feea872178452e5302314a5/crates/primitives/src/receipt.rs#L57-L62
-#[derive(Debug, RlpEncodable, PartialEq, Clone)]
+#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
```

Comment:
This will have to be `no_std` for the pallet, which will be a pain later...

Reply:
Smith did it, so I thought he tested it with a pallet

Reply:
Sounds good